### DISCUSSION TITLE
ThreadDataStore alignment

### DISCUSSION THREAD

**Discussion by Liemarzac** [dougbinks/enkiTS#82] [General]
Hi. I don't think this line is required: `char prevent_false_Share[ enki::gc_CacheLineSize - sizeof(std::atomic<ThreadState>) - sizeof(semaphoreid_t*) ];`
The compiler will pad the structure automatically so that every instance in an array will be aligned on `enki::gc_CacheLineSize` because alignas was used.

**Comment by dougbinks** [comment]
Thanks for spotting, and I agree. I can't recall why I added this - potentially just habit from before alignment was easy to do cross platform.

**Comment by dougbinks** [comment]
I resolved this in:

https://github.com/dougbinks/enkiTS/commit/67fe56e51ae0681135c43ee8ff8c7d262f024c8a

**Reply by Liemarzac** [reply]
I'm sorry, I might have been to quick to suggest to remove the padding. I've just used enkiTS without your explicit padding in a codebase where MSVC is set with a warning level 4 and it does trigger warning 4324: "structure was padded due to alignment specifier"
It might explain why you padding explicitly to start with. I find this warning not useful but I thought I should still tell you about it. I personally just silenced it. My biggest problem with the manual padding is that it's easy to get a calculation which will end up padding with a negative number if the sum of the size of all members gets bigger than the cache line itself, and 
 wrap around a uint32 and create a massive number.

**Comment by dougbinks** [comment]
Ah! I think that might explain why I added it. Thanks again, will revert that change and add an explanation.
